server/src/main/java/org/jboss/as/server/deployment/DeploymentAddHandler.java

```diff
@@ -219,8 +219,17 @@ private static String getRuntimeName(String name, ModelNode deployment) {
             try {
                 transformed = deploymentTransformer == null ? in : deploymentTransformer.transform(in, name);
                 hash = contentRepository.addContent(transformed);
-            } catch (IOException e) {
+            } catch (IOException e) { // fail hard on IO error
                 throw createFailureException(e.toString());
+            } catch(Throwable t) {  // for other errors, return original file stream
```

Comment:
I assume what we are really trying to catch is a RuntimeException thrown by EclipseTransformer, not a java.lang.Error. So this catch should be limited to RuntimeException.

Beyond a general code standard thing of avoiding suppressing Error, this matters because if we bypass trying again for deployments that use DeploymentAttributes.CONTENT_INPUT_STREAM_INDEX, then for those we are going to have to rethrow the exception.

Reply:
Also, I think this catch should be for a try block that only includes L220.  We want to react to failures in deploymentTransformer.transform, not to anything that happens with L221's contentRepository.addContent.

Creating a method for the L220 and the try/catch would be cleaner looking.